crates/release_plz_core/src/command/release.rs

```diff
@@ -741,28 +779,59 @@ fn registry_indexes(
 
     let mut registry_indexes = registry_urls
         .into_iter()
-        .map(|(registry, u)| {
-            if u.to_string().starts_with("sparse+") {
-                SparseIndex::from_url_with_hash_kind(u.as_str(), hash_kind).map(CargoIndex::Sparse)
-            } else {
-                GitIndex::from_url_with_hash_kind(&format!("registry+{u}"), hash_kind)
-                    .map(CargoIndex::Git)
-            }
-            .map(|index| CargoRegistry {
-                name: Some(registry),
-                index,
-            })
-        })
-        .collect::<Result<Vec<CargoRegistry>, crates_index::Error>>()?;
+        .map(|(registry, u)| get_cargo_registry(hash_kind, registry, u))
+        .collect::<anyhow::Result<Vec<CargoRegistry>>>()?;
     if registry_indexes.is_empty() {
         registry_indexes.push(CargoRegistry {
             name: None,
             index: CargoIndex::Git(GitIndex::new_cargo_default()?),
+            fallback_index: None,
         });
     }
     Ok(registry_indexes)
 }
 
+fn get_cargo_registry(
+    hash_kind: &crates_index::HashKind,
+    registry: String,
+    u: Url,
+) -> anyhow::Result<CargoRegistry> {
+    let fallback_hash = try_get_fallback_hash_kind(hash_kind);
+
+    let (maybe_primary_index, maybe_fallback_index) = if u.to_string().starts_with("sparse+") {
+        let index_url = u.as_str();
+        let maybe_primary =
+            SparseIndex::from_url_with_hash_kind(index_url, hash_kind).map(CargoIndex::Sparse);
+        let maybe_fallback = fallback_hash.map(|hash_kind| {
+            SparseIndex::from_url_with_hash_kind(index_url, &hash_kind).map(CargoIndex::Sparse)
+        });
+
+        (maybe_primary, maybe_fallback)
+    } else {
+        let index_url = format!("registry+{u}");
+        let maybe_primary =
+            GitIndex::from_url_with_hash_kind(&index_url, hash_kind).map(CargoIndex::Git);
```

Comment:
`(_, Some(Err(e))` worries me. Maybe it can happen that the primary index works just fune but we fail because of weird errors on the fallback?